xdr/next/Stellar-contract.x

```diff
@@ -71,16 +71,115 @@ enum SCStatic
 enum SCStatusType
 {
     SST_OK = 0,
-    SST_UNKNOWN_ERROR = 1
+    SST_UNKNOWN_ERROR = 1,
+    SST_HOST_VALUE_ERROR = 2,
+    SST_HOST_OBJECT_ERROR = 3,
+    SST_HOST_FUNCTION_ERROR = 4,
+    SST_HOST_STORAGE_ERROR = 5,
+    SST_HOST_CONTEXT_ERROR = 6,
+    SST_VM_ERROR = 7
     // TODO: add more
 };
 
+enum SCHostValErrorCode
+{
+    HOST_VALUE_UNKNOWN_ERROR = 0,
+    RESERVED_TAG_VALUE = 1,
+    UNEXPECTED_VAL_TYPE = 2,
+    U63_OUT_OF_RANGE = 3,
+    U32_OUT_OF_RANGE = 4,
+    STATIC_UNKNOWN = 5,
+    MISSING_OBJECT = 6,
+    SYMBOL_TOO_LONG = 7,
+    SYMBOL_BAD_CHAR = 8,
+    SYMBOL_CONTAINS_NON_UTF8 = 9,
+    BITSET_TOO_MANY_BITS = 10,
+    STATUS_UNKNOWN = 11
+};
+
+enum SCHostObjErrorCode
+{
+    HOST_OBJECT_UNKNOWN_ERROR = 0,
+    UNKNOWN_HOST_OBJECT_REFERENCE = 1,
+    UNEXPECTED_HOST_OBJECT_TYPE = 2,
+    OBJECT_HANDLE_EXCEEDS_U32_MAX = 3,
+    ACCESSING_HOST_OBJECT_OUT_OF_BOUND = 4,
+    VEC_INDEX_OUT_OF_BOUND = 5,
+    VEC_INDEX_OVERFLOW = 6,
+    VEC_VALUE_NOT_EXIST = 7,
```

Comment:
When does a vec value not exist?

Reply:
Changed to `HOST_OBJECT_VEC_ELEMENT_NOT_EXIST`.

Reply:
What does it mean that a vec element doesn't exist though? Is that the same thing as index out of bounds?

Reply:
These currently happens in `vec_front`, `vec_back` and `vec_pop`, when there is no element in the vec. Should I rename it to `HOST_OBJECT_VEC_IS_EMPTY` instead?

Reply:
I would expect the error to be the same that the index is out of bound. Distinguishing between these cases feels like an implementation details is leaking into the protocol. If we need a unique error for some reason then "vec is empty" is clearer.

Reply:
Ok, I've removed the `HOST_OBJECT_VEC_IS_EMPTY` case. Will just treat it same as "out of bound".